### ISSUE TITLE
Compatibility with Doctrine DBAL 3

### ISSUE THREAD

**Issue by DominicDetta** [itsgoingd/clockwork#678]
With Doctrine DBAL 3.0.0 the current `DBALDataSource` does not work because the method `LoggerChain::addLogger`  has been removed and the class `SQLLogger` is deprecated.
The new way to implement logging is using [Middleware](https://github.com/doctrine/dbal/pull/4967).

The error obtained: 

```php
Fatal error: Uncaught Error: Call to undefined method Doctrine\DBAL\Logging\LoggerChain::addLogger() in /var/www/html/vendor/itsgoingd/clockwork/Clockwork/DataSource/DBALDataSource.php:39
```

I will try to create a working example in my project and create a PR if possible

**Comment by DominicDetta** [comment]
This error appears only if there is already a `SQLLogger` configured

**Comment by itsgoingd** [comment]
I've made a PR to support for Doctrine 3+ (https://github.com/itsgoingd/clockwork/pull/720), to be released in Clockwork 5.3.
